### ISSUE TITLE
bug: Error [ERR_REQUIRE_ESM]: require() of ES Module 

### ISSUE THREAD

**Issue by erdenetsogta** [saadeghi/daisyui#2498]
### What version of daisyUI are you using?

v4.0.1

### Which browsers are you seeing the problem on?

Chrome

### Reproduction URL

http://localhost:4200/

### Describe your issue

```
./app/global.css.webpack[javascript/auto]!=!../../../node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[13].oneOf[12].use[2]!../../../node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[13].oneOf[12].use[3]!./app/global.css
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/erdenetsogt/Documents/playground/keshava/node_modules/culori/src/index.js from /Users/erdenetsogt/Documents/playground/keshava/node_modules/daisyui/src/theming/functions.js not supported.
Instead change the require of index.js in /Users/erdenetsogt/Documents/playground/keshava/node_modules/daisyui/src/theming/functions.js to a dynamic import() which is available in all CommonJS modules.
```

**Comment by saadeghi** [comment]
Please provide a reproduction repo so I can run your project.

**Comment by lingster** [comment]
I have the same issue, reverting to v3 of daisy UI solves this. 

I think the problem is with:

```
const { toGamut, interpolate, wcagContrast } = require("culori")
```
should this import instead?

**Comment by saadeghi** [comment]
Are you importing `daisyui/src/theming/functions.js` file inside your project for some reason?  
Can you share a reproduction repo?

**Comment by epicbytes** [comment]
Same problem after updating to version 4.

```json
{
"daisyui": "4.0.7",
"vite": "4.5.0",
"solid-js": "1.8.5"
}
```
daisy connected to project as a plugin:

```js
module.exports = {
 plugins: [require("daisyui")]
}
```

```bash
 [solid] [BABEL] /Volumes/WORK/generatorv2/frontend/src/index.tsx: require() of ES Module /Volumes/WORK/generatorv2/frontend/node_modules/culori/src/index.js from /Volumes/WORK/generatorv2/frontend/node_modules/daisyui/src/theming/functions.js not supported.
    Instead change the require of index.js in /Volumes/WORK/generatorv2/frontend/node_modules/daisyui/src/theming/functions.js to a dynamic import() which is available in all CommonJS modules. (While processing: "/Volumes/WORK/generatorv2/frontend/node_modules/typewind/dist/babel.js")
    file: /Volumes/WORK/generatorv2/frontend/src/index.tsx
    error during build:
    Error [ERR_REQUIRE_ESM]: require() of ES Module /Volumes/WORK/generatorv2/frontend/node_modules/culori/src/index.js from /Volumes/WORK/generatorv2/frontend/node_modules/daisyui/src/theming/functions.js not supported.

```

**Comment by franck-lynn** [comment]
same the problem .
```
[1] [1] failed to load config from F:\working\study\alchemilla\client\vite.config.ts
[1] [1] error when starting dev server:
[1] [1] Error [ERR_REQUIRE_ESM]: require() of ES Module F:\node\node_modules\culori\src\index.js from F:\node\node_modules\daisyui\src\theming\functions.js not supported.
[1] [1] Instead change the require of index.js in F:\node\node_modules\daisyui\src\theming\functions.js to a dynamic import() which is available in all CommonJS modules.
[1] [1]     at _require.extensions.<computed> [as .js] (file:///F:/node/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:66343:17)
[1] [1]     at Object.<anonymous> (F:\node\node_modules\daisyui\src\theming\functions.js:4:48)
[1] [1]     at _require.extensions.<computed> [as .js] (file:///F:/node/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:66343:17)
```
it should be 'culori' problem?

**Comment by saadeghi** [comment]
I'm not sure 🤔  
If someone provides a reproduction repo, it would be helpful to find the problem.

**Comment by nader-eloshaiker** [comment]
Have a related issue. Using `vite` and getting `[vite] Internal server error: [postcss] Cannot read properties of undefined (reading 'l')` when upgrading to v4.0.7 - 4.0.9.

**Comment by TurtleCode84** [comment]
I'm currently experiencing this issue in [TurtleCode84/turtlecode.tech](https://github.com/turtlecode84/turtlecode.tech) after updating `daisyui` from 3.9.4 to 4.0.5 in [this Actions run](https://github.com/TurtleCode84/turtlecode.tech/actions/runs/6876886953), not sure if the repo is helpful.

**Comment by saadeghi** [comment]
@TurtleCode84 @franck-lynn @epicbytes @lingster @erdenetsogta 
Can you please try uninstalling and installing daisyui?
```
npm remove daisyui
npm i daisyui@latest
```
This seems to fix the issue. I'm not sure why…

**Comment by TurtleCode84** [comment]
> Can you please try uninstalling and installing daisyui?

> unfortunately it does not seem to solve it (same issue as above)

Can confirm that un/reinstalling did not resolve the issue.

**Comment by saadeghi** [comment]
It must be fixed now in `4.3.1`  
Sorry about the issue.

**Comment by erdenetsogta** [comment]
> It must be fixed now in `4.3.1` Sorry about the issue.

Yes. It did fix the issue. 

https://github.com/erdenetsogta/nx-daisy
`npm i`
`npx nx serve demo`

Thank you.

**Comment by droganov** [comment]
Fixed parcel issue with alias in package.json:

```json
"alias": {
    "culori/require": "culori"
  }
  ```
